### PR TITLE
User settings spec

### PR DESCRIPTION
**Pull request by tonytamps**
Initial draft of user settings spec

I read all the other specs and stole what seemed relevant to user settings.

Am a little unsure about how to write use cases or data models for the user settings so throw me a bone if you have a better idea.

### CODE REVIEW

File: doc/specs/userSettings.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+**Actor:** User.
+
+**Brief:**  A user can change their own settings to interact with the site differently.
+
+**Trigger:** User wants to change a setting.
+
+**Preconditions:**
+
+ - The user can only change their own settings.
```
Comment (ghost) [inline doc/specs/userSettings.md]: This is a BR.  Also it's not accurate because mods+ can edit other user settings. There's a permission for this.

Comment (tonytamps) [inline doc/specs/userSettings.md]: Oooh sweet, I'll make changes to the entire doc to better reflect this.

File: doc/specs/userSettings.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# User Settings
+
+## Metadata
+
+Author: Tick
```
Comment (ghost) [inline doc/specs/userSettings.md:5]: We aren't using metadata or changelog.

File: doc/specs/userSettings.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+1. Business rules
+1. Use cases
+
+## Motivation
+
+Provides a means for users of the site to configure and personalise their usage of the site.
+
+## Business rules.
+
+ - A user can only edit the own settings
```
Comment (ghost) [inline doc/specs/userSettings.md:25]: the -> their

File: doc/specs/userSettings.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+ - Administrators with the right permissions can edit another user's settings.
```
Comment (ghost) [inline doc/specs/userSettings.md:26]: What about users with the settings admin permission trying to edit the settings of a user above them? i.e. a mod trying to edit an admin's settings.

File: doc/specs/userSettings.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+Settings can be configured for:
+
+* **Site Appearance**: Stylesheets and fonts.
+
+* **Torrents**: Appearance of search results and cover art.
+
+* **Community**: Forums, inbox, avatars and Rippy.
```
Comment (ghost) [inline doc/specs/userSettings.md:34]: Rippy is internal.

Edit: It looks like we'll be using a more generic version of Rippy with this, called "site mascot".

File: doc/specs/userSettings.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+## Business rules.
+
+ - A user can only edit the own settings
+ - Administrators with the right permissions can edit another user's settings.
+
+Settings can be configured for:
+
+* **Site Appearance**: Stylesheets and fonts.
+
+* **Torrents**: Appearance of search results and cover art.
```
Comment (ghost) [inline doc/specs/userSettings.md:32]: Cover art is music specific. Try to keep it more generic.

File: doc/specs/userSettings.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+* **Community**: Forums, inbox, avatars and Rippy.
+
+* **Notifications**: News announcements, inbox messages, announcements, forum threads, torrent notifications, etc...
+
+* **Personal**: Avatar, Last.fm and title.
```
Comment (ghost) [inline doc/specs/userSettings.md:38]: Last.FM is music specific.

File: doc/specs/userSettings.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+ - The user's settings are updated.
+ - The user's log is updated.
+
+**Basic Flow:**
+
+ 1. The user navigates to their own settings page.
+ 2. The user changes the setting to the desired value.
+ 4. The user saves the changes to their settings.
+
+### Administrator changes another user's setting
```
Comment (ghost) [inline doc/specs/userSettings.md:65]: Not necessarily an administrator. I'd say "A user with the settings management permission" or something along those lines.

**Comment by ghost** [comment]
In terms of the data model, user settings can be a subdocument on the user schema. We have to discuss how to store it, because I doubt we'll be storing the paranoia as a serialized string. @ajax-wcd

File: doc/specs/userSettings.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+## Table of contents
+
+1. Motivation
+1. Business rules
+1. Use cases
+
+## Motivation
+
+Provides a means for users of the site to configure and personalise their usage of the site.
```
Comment (Malvar) [inline doc/specs/userSettings.md:21]: personalise -> personalize

**Comment by ghost** [comment]
Issue #63

**Pull request closed without merging**